src/lib.rs

```diff
@@ -239,14 +353,15 @@ pub mod write {
     /// - `serial_number.len() > 20`
     /// - `issuer.len() > 64`
     /// - `subject.len() > 64`
-    pub fn tbs_certificate<'a, W: Write + 'a, Alg, PKI>(
+    pub fn tbs_certificate<'a, W: Write + 'a, Alg, PKI, O: Oid + 'a>(
         serial_number: &'a [u8],
         signature: &'a Alg,
         issuer: &'a str,
         not_before: DateTime<Utc>,
         not_after: Option<DateTime<Utc>>,
         subject: &'a str,
         subject_pki: &'a PKI,
+        exts: &'a [Extension<'a, O>],
```

Comment:
I considered making this `Option<_>` but it didn't solve the issue of needing to provide some type for `O: Oid`, so I decided to just stick with "empty slice means no extensions". If we later add some Certificate structs, we can provide a default type there.